clusterloader2/cmd/clusterloader.go

```diff
@@ -31,38 +32,58 @@ import (
 )
 
 var (
-	kubeConfigPath = flag.String("kubeconfig", "", "path to the kubeconfig file")
-	testConfigPath = flag.String("testconfig", "", "path to the test config file")
+	clusterConfig  api.ClusterConfig
+	testConfigPath string
 )
 
-func validateFlags() []error {
+func initClusterFlags() {
+	flag.StringVar(&clusterConfig.KubeConfigPath, "kubeconfig", "", "Path to the kubeconfig file")
+	flag.IntVar(&clusterConfig.NodeCount, "nodes", -1, "Number of nodes in the cluster")
+}
+
+func validateClusterFlags() []error {
 	errList := make([]error, 0)
-	if *kubeConfigPath == "" {
+	if clusterConfig.KubeConfigPath == "" {
 		errList = append(errList, fmt.Errorf("no kubeconfig path specified"))
 	}
-	if *testConfigPath == "" {
+	return errList
+}
+
+func initFlags() {
+	flag.StringVar(&testConfigPath, "testconfig", "", "path to the test config file")
+	initClusterFlags()
+}
+
+func validateFlags() []error {
+	errList := make([]error, 0)
+	if testConfigPath == "" {
 		errList = append(errList, fmt.Errorf("no test config path specified"))
 	}
+	errList = append(errList, validateClusterFlags()...)
 	return errList
 }
 
 func main() {
 	defer glog.Flush()
+	initFlags()
 	flag.Parse()
 	if errList := validateFlags(); len(errList) > 0 {
 		glog.Fatalf("Parsing flags error: %v", errors.NewAggregate(errList).Error())
 	}
 
-	f, err := framework.NewFramework(*kubeConfigPath, filepath.Dir(*testConfigPath))
+	if testConfigPath == "" {
```

Comment:
I think this condition is already checked in validateFlags() above.

Reply:
Good catch. Redundant check removed.